### DISCUSSION TITLE
YouTube++ and CarPlay

### DISCUSSION THREAD

**Discussion by jiggers84** [qnblackcat/uYouPlus#90] [Ideas]
After talking with the dev for "carplayenable". They told me it would be possible to make this work with an  unjailbroken device with some work. 

"The tweak requires code injection into system processes (CarPlay) as well as injection into every app that is going to be forced onto the CarPlay display (to adapt the app’s window to the CarPlay unit screen size) 

It would be possible, but complex, to modify individual apps to work with CarPlay on non-jb ios. If you have extensive experience with runtime modification development and an enterprise development certificate I can describe what is needed"

Using the new update to sideloadly which will allow you inject dylib & deb with substrate combined with an ipa. Is there anyway to combine all of this together, and allow YouTube++ to work with CarPlay as a new feature? Is this plausible?

**Comment by qnblackcat** [comment]
This is something beyond my knowledge

**Comment by BandarHL** [comment]
Even if you hook CarPlay framework, will not work because in sideloaded IPA you re-sign it with different certificate and **Entitlements** and with CarPlay you must have their Entitlements and it’s not easy to git it, because you must have paid developer account and application on AppStore and then you submit request to apple to give you the Entitlements.
As you can see it’s mostly impossible to do it, and don’t forget most of the uYou++ users use free signed (7days) like Altstore.

**Reply by BandarHL** [reply]
https://developer.apple.com/documentation/carplay/requesting_carplay_entitlements
